### PR TITLE
Add support of compression to federation responses

### PR DESCRIPTION
**Pull request by iamayushanand**
Closes Issue #13415

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Ayush Anand [iamayushanand@gmail.com](mailto:iamayushanand@gmail.com)

### CODE REVIEW

File: changelog.d/13537.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Compression is now supported for federation responses.
```
Comment (squahtx) [inline changelog.d/13537.bugfix:1]: We usually start changelog entries with a verb, eg.
```suggestion
Add support for compression to federation responses.
```

**Review by squahtx — COMMENTED** [review]
Looks good!

One minor suggestion:

**Comment by clokep** [comment]
I think we need to update the documentation (or backout the changes from #13221).

**Comment by iamayushanand** [comment]
> I think we need to update the documentation (or backout the changes from #13221).

yup, so I need to make a seperate PR for that? or should I do the documentation changes too in this PR and then create another changelog for doc as well?

**Comment by clokep** [comment]
> yup, so I need to make a seperate PR for that? or should I do the documentation changes too in this PR and then create another changelog for doc as well?

You can do it all in this PR with the single changelog file you have already.

**Review by squahtx — APPROVED** [review]
Looks good! Thanks for keeping the documentation up to date.

**Pull request merged**